src/flint/fmpq_mpoly.jl

```diff
@@ -474,6 +474,73 @@ function gcd(a::fmpq_mpoly, b::fmpq_mpoly)
    return z
 end
 
+################################################################################
+#
+#   Factorization and Square Root
+#
+################################################################################
+
+function Base.convert(::Type{Fac{fmpq_mpoly}}, fac::fmpq_mpoly_factor)
+   R = fac.parent
+   dic = Dict{fmpq_mpoly, Int}()
+   for i in 0:fac.num-1
+      f = R()
+      # use get_base instead of swap_base if convert is to preserve input
```

Comment:
@rfourquet Is it ok if this function destroys its input? I just didn't want make another useless copy of the polynomials...

Reply:
Why did you implement a `convert` method? Or was it there before?

Reply:
if I implement a convert function, there is less copy+paste and fewer lines of code

Reply:
The problem with `convert` is that it will be called implicitely without the users knowledge (for example in array or dictionary assignments), making the destructive behavior rather unfortunate.

How about implementing `(::Factor{fmpq_poly})(::fmpz_poly_factor)`? Or `convert_and_destroy`, something that indicates that the input will be garbage.

Reply:
Yes I agree with @thofma